examples/gno.land/p/demo/paginate/paginate.gno

```diff
@@ -12,22 +19,116 @@ type renderer interface {
 // TODO this function belongs somewhere convenient.
 // escape a URL for markdown.
 func escMDURL(s string) string {
-	// XXX implement
+	schars := map[string]string{
+		"%":  "%25",
+		" ":  "%20",
+		"!":  "%21",
+		"\"": "%22",
+		"$":  "%24",
+		"&":  "%26",
+		"'":  "%27",
+		"(":  "%28",
+		")":  "%29",
+		"*":  "%2A",
+		"+":  "%2B",
+		",":  "%2C",
+		";":  "%3B",
+		"<":  "%3C",
+		">":  "%3E",
+		"?":  "%3F",
+		"@":  "%40",
+		"[":  "%5B",
+		"\\": "%5C",
+		"]":  "%5D",
+		"^":  "%5E",
+		"_":  "%5F",
+		"`":  "%60",
+		"{":  "%7B",
+		"|":  "%7C",
+		"}":  "%7D",
+		"~":  "%7E",
+		"\n": "%0A",
+	}
+	for k, v := range schars {
+		s = strings.Replace(s, k, v, -1)
+	}
+	return s
 }
 
-var itoa = strconv.Itoa
+func navigation(numPages int, current int, reverse bool, pageFn pageFn) string {
+	str := ""
+
+	i := 1
+	for ; i <= PAGINATE_GAP && i < current; i++ {
```

Comment:
```suggestion
	for i := 1 ; i <= PAGINATE_GAP && i < current; i++ {
```

Reply:
```suggestion
	upperLimit := min(PaginateGap + 1, current)
	for i := 1; i < upperLimit; i++ {
```

easier to read

Reply:
keep the index along the way over the range that's why I don't declare it in the scope of a loop.

For the `upperLimit` variable I tried to do it in a first way but didn't see any `math.Min` do you think it is valuable to do a PR to add it for the numeric values ?

Reply:
inside the Math package

Reply:
no, something like this is better

```go
func min(a, b int) int {...}
```